classes/task/get_meeting_recordings.php

```diff
@@ -109,17 +104,19 @@ public function execute() {
             $zoomrecordings = $service->get_user_recordings($hostid, $from, $to);
 
             foreach ($zoomrecordings as $recordingid => $recording) {
-                if (isset($localrecordings[$recording->meetinguuid][$recordingid])) {
-                    mtrace('Recording id: ' . $recordingid . ' exists...skipping');
-                    continue;
-                }
-
                 if (empty($meetings[$recording->meetingid])) {
                     // Skip meetings that are not in Moodle.
                     mtrace('Meeting id: ' . $recording->meetingid . ' does not exist...skipping');
                     continue;
                 }
 
+                $zoom = $meetings[$recording->meetingid];
+
+                if (isset($localrecordings[$recording->meetinguuid][$recordingid])) {
+                    mtrace('Recording id: ' . $recordingid . ' exists...skipping');
+                    continue;
```

Comment:
In case we load data for a recording and it has a more accurate recording type, update the local recording data to have the better recording type value.
```suggestion
                    mtrace('Recording id: ' . $recordingid . ' exists...skipping');
                    $localrecording = $localrecordings[$recording->meetinguuid][$recordingid];

                    if ($localrecording->recordingtype !== $zoom->recordingtype) {
                        $updatemeeting = (object) [
                            'id' => $localrecording->id,
                            'recordingtype' => $zoom->recordingtype,
                        ];
                        $DB->update_record('zoom_meeting_recordings', $updatemeeting);
                    }
                    continue;
```